subsys/shell/Kconfig.backends

```diff
@@ -167,6 +167,92 @@ source "subsys/logging/Kconfig.template.log_config"
 
 endif #SHELL_BACKEND_RTT
 
+config SHELL_BACKEND_TELNET
+	bool "Enable TELNET backend."
+	help
+	  Enable TELNET backend.
+
+if SHELL_BACKEND_TELNET
+
+config SHELL_BACKEND_TELNET_PORT
+	int "Telnet port number"
+	default 23
+	help
+	  This option is used to configure on which port telnet is going
+	  to be bound.
+
+config SHELL_BACKEND_TELNET_LINE_BUF_SIZE
+	int "Telnet line buffer size"
+	default 80
+	help
+	  This option can be used to modify the size of the buffer storing
+	  shell output line, prior to sending it through the network.
+	  Of course an output line can be longer than such size, it just
+	  means sending it will start as soon as it reaches this size.
+	  It really depends on what type of output is expected.
+	  A lot of short lines: better reduce this value. On the contrary,
+	  raise it.
+
+config SHELL_BACKEND_TELNET_SEND_TIMEOUT
+	int "Telnet line send timeout"
+	default 100
+	help
+	  This option can be used to modify the duration of the timer that kick
+	  in when a line buffer is not empty but did not yet meet the line feed.
+
+config SHELL_BACKEND_TELNET_SUPPORT_COMMAND
+	bool "Add support for telnet commands (IAC) [Experimental]"
+	help
+	  Current support is so limited it's not interesting to enable it.
+	  However, if proven to be needed at some point, it will be possible
+	  to extend such support.
+
+module = SHELL_BACKEND_TELNET
+default-timeout = 100
+source "subsys/shell/Kconfig.template.shell_log_queue_timeout"
+
+default-size = 10
+source "subsys/shell/Kconfig.template.shell_log_queue_size"
+
+choice
+	prompt "Initial log level limit"
+	default SHELL_BACKEND_TELNET_LOG_LEVEL_DEFAULT
+
+config SHELL_BACKEND_TELNET_LOG_LEVEL_DEFAULT
+	bool "System limit (LOG_MAX_LEVEL)"
+
+config SHELL_BACKEND_TELNET_LOG_LEVEL_DBG
+	bool "Debug"
+
+config SHELL_BACKEND_TELNET_LOG_LEVEL_INF
+	bool "Info"
+
+config SHELL_BACKEND_TELNET_LOG_LEVEL_WRN
+	bool "Warning"
+
+config SHELL_BACKEND_TELNET_LOG_LEVEL_ERR
+	bool "Error"
+
+config SHELL_BACKEND_TELNET_LOG_LEVEL_NONE
+	bool "None"
+
+endchoice
+
+config SHELL_BACKEND_TELNET_LOG_LEVEL
```

Comment:
this could be renamed to `SHELL_TELNET_INIT_LOG_LEVEL` and `SHELL_BACKEND_TELNET_LOG_LEVEL_<lvl>` -> `SHELL_TELNET_INIT_LOG_LEVEL_<lvl>`. RTT is using that naming https://github.com/zephyrproject-rtos/zephyr/blob/766edd449d6c9b654ca49aa21390b9514e9e6aa6/subsys/shell/Kconfig.backends#L155 and serial could also be changed to use that since it's more descriptive.